packages/@ember/-internals/runtime/lib/ext/rsvp.ts

```diff
@@ -0,0 +1,75 @@
+import * as RSVP from 'rsvp';
+import { _backburner, _rsvpErrorQueue } from '@ember/runloop';
+import { getDispatchOverride } from '@ember/-internals/error-handling';
+import { assert } from '@ember/debug';
+
+RSVP.configure('async', (callback: unknown, promise: unknown) => {
+  _backburner.schedule('actions', null, callback, promise);
+});
+
+RSVP.configure('after', (cb: unknown) => {
+  _backburner.schedule(_rsvpErrorQueue, null, cb);
+});
+
+RSVP.on('error', onerrorDefault);
+
+export function onerrorDefault(reason: unknown) {
+  let error = errorFor(reason);
+  if (error) {
+    let overrideDispatch = getDispatchOverride();
+    if (overrideDispatch) {
+      overrideDispatch(error);
+    } else {
+      throw error;
+    }
+  }
+}
+
+interface ReasonWithErrorThrown {
+  errorThrown: unknown;
+}
+
+interface ReasonWithName {
+  name: unknown;
+}
+
+interface UnrecognizedURLError {
+  name: 'UnrecognizedURLError';
+  message: string;
+}
+
+function errorFor(reason: unknown) {
+  if (!reason) return;
+
+  let withErrorThrown = reason as ReasonWithErrorThrown;
+  if (withErrorThrown?.errorThrown) {
```

Comment:
Optional chaining not needed here given the `!reason` check on line 42.

```suggestion
  if (withErrorThrown.errorThrown) {
```